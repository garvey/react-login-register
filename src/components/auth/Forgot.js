import React from 'react';
import { Link } from 'react-router-dom';
import useForm from 'react-hook-form';
import Logo from '../../images/logo.png';
import { FiAlertCircle } from 'react-icons/fi';
import {
  Col,
  Form,
  FormGroup,
  Label,
  Button,
  Container,
  Card,
  CardBody,
  Alert
} from 'reactstrap';

function Forgot(props) {
  const { register, handleSubmit, errors } = useForm();

  const onSubmit = res => {
    console.log('Password Data:', res);
  };

  return (
    <Container>
      <Col
        xs={12}
        md={{ size: 6, offset: 3 }}
        lg={{ size: 4, offset: 4 }}
        className="my-auto mx-auto"
      >
        <Form onSubmit={handleSubmit(onSubmit)}>
          <div className="text-center pt-4 pb-4">
            <Link to="/">
              <img src={Logo} alt="Company Logo" className="img-fluid logo" />
            </Link>
          </div>
          <Card className="login">
            <CardBody>
              <h5>Forgot your password?</h5>
              <small>
                Enter your email below and we'll reset your password.
              </small>
              <FormGroup className="pt-5">
                <Label>Your Email Address</Label>

                <input
                  name="email"
                  ref={register({
                    required: 'Required email address',
                    pattern: {
                      value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                      message: 'invalid email address'
                    }
                  })}
                />
              </FormGroup>

              {errors.email && errors.email.message && (
                <Alert color="info">
                  <FiAlertCircle /> Please enter a valid email address
                </Alert>
              )}

              <Button outline color="success" block type="submit">
                Reset Password
              </Button>

              <FormGroup>
                <p className="pt-4 text-center">
                  Back to <Link to="/login">Login</Link>
                </p>
              </FormGroup>
            </CardBody>
          </Card>
        </Form>
      </Col>
    </Container>
  );
}

export default Forgot;
