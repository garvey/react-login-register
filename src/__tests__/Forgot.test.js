import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Forgot from '../components/auth/Forgot';

it('Should render Component', () => {
  const wrapper = mount(
    <MemoryRouter>
      <Forgot />
    </MemoryRouter>
  );
  expect(wrapper.debug()).toMatchSnapshot();
});
