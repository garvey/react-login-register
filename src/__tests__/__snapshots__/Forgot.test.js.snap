// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Component 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Forgot>
      <Container tag=\\"div\\">
        <div className=\\"container\\">
          <Col xs={12} md={{...}} lg={{...}} className=\\"my-auto mx-auto\\" tag=\\"div\\" widths={{...}}>
            <div className=\\"my-auto mx-auto col-12 col-md-6 offset-md-3 col-lg-4 offset-lg-4\\">
              <Form onSubmit={[Function]} tag=\\"form\\">
                <form onSubmit={[Function]} className=\\"\\">
                  <div className=\\"text-center pt-4 pb-4\\">
                    <Link to=\\"/\\">
                      <LinkAnchor href=\\"/\\" navigate={[Function: navigate]}>
                        <a href=\\"/\\" onClick={[Function: onClick]}>
                          <img src=\\"logo.png\\" alt=\\"Company Logo\\" className=\\"img-fluid logo\\" />
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                  <Card className=\\"login\\" tag=\\"div\\">
                    <div className=\\"login card\\">
                      <CardBody tag=\\"div\\">
                        <div className=\\"card-body\\">
                          <h5>
                            Forgot your password?
                          </h5>
                          <small>
                            Enter your email below and we&#39;ll reset your password.
                          </small>
                          <FormGroup className=\\"pt-5\\" tag=\\"div\\">
                            <div className=\\"pt-5 form-group\\">
                              <Label tag=\\"label\\" widths={{...}}>
                                <label htmlFor={[undefined]} className=\\"\\">
                                  Your Email Address
                                </label>
                              </Label>
                              <input name=\\"email\\" />
                            </div>
                          </FormGroup>
                          <Button outline={true} color=\\"success\\" block={true} type=\\"submit\\" tag=\\"button\\">
                            <button type=\\"submit\\" className=\\"btn btn-outline-success btn-block\\" onClick={[Function: bound onClick]} aria-label={{...}}>
                              Reset Password
                            </button>
                          </Button>
                          <FormGroup tag=\\"div\\">
                            <div className=\\"form-group\\">
                              <p className=\\"pt-4 text-center\\">
                                Back to 
                                <Link to=\\"/login\\">
                                  <LinkAnchor href=\\"/login\\" navigate={[Function: navigate]}>
                                    <a href=\\"/login\\" onClick={[Function: onClick]}>
                                      Login
                                    </a>
                                  </LinkAnchor>
                                </Link>
                              </p>
                            </div>
                          </FormGroup>
                        </div>
                      </CardBody>
                    </div>
                  </Card>
                </form>
              </Form>
            </div>
          </Col>
        </div>
      </Container>
    </Forgot>
  </Router>
</MemoryRouter>"
`;
